sklearn/linear_model/base.py

```diff
@@ -112,6 +112,19 @@ def center_data(X, y, fit_intercept, normalize=False, copy=True,
     return X, y, X_mean, y_mean, X_std
 
 
+def rescale_data(X, y, sample_weight):
+    """Rescale data so as to support sample_weight"""
+    n_samples = X.shape[0]
+    sample_weight = sample_weight * np.ones(n_samples)
```

Comment:
This function is used in ridge.py as well. For `Ridge`, does it make sense to have a scaler `sample_weight`? Did some simple tests, it did not change the output. If scalar `sample_weight` can go away in both cases, it will be easier to write this function.